src/core/include/mp-units/cartesian_vector.h

```diff
@@ -135,27 +145,29 @@ class cartesian_vector : public detail::cartesian_vector_iface {
   T _coordinates_[3];
   using value_type = T;
 
+
+  cartesian_vector() = default;
   cartesian_vector(const cartesian_vector&) = default;
   cartesian_vector(cartesian_vector&&) = default;
   cartesian_vector& operator=(const cartesian_vector&) = default;
   cartesian_vector& operator=(cartesian_vector&&) = default;
 
   template<typename... Args>
-    requires(... && std::constructible_from<T, Args>)
+    requires(sizeof...(Args) <= 3) && (... && std::constructible_from<T, Args>)
   constexpr explicit(!(... && std::convertible_to<Args, T>)) cartesian_vector(Args&&... args) :
       _coordinates_{static_cast<T>(std::forward<Args>(args))...}
   {
   }
 
   template<typename U>
-    requires std::constructible_from<T, U>
+    requires std::constructible_from<T, const U&>
   constexpr explicit(!std::convertible_to<U, T>) cartesian_vector(const cartesian_vector<U>& other) :
       _coordinates_{static_cast<T>(other[0]), static_cast<T>(other[1]), static_cast<T>(other[2])}
   {
   }
 
   template<typename U>
-    requires std::constructible_from<T, U>
+    requires std::constructible_from<T, U&&>
   constexpr explicit(!std::convertible_to<U, T>) cartesian_vector(cartesian_vector<U>&& other) :
       _coordinates_{static_cast<T>(std::move(other[0])), static_cast<T>(std::move(other[1])),
                     static_cast<T>(std::move(other[2]))}
```

Comment:
Let's remove all of those. This will result in an aggregate type (such as `std::array`), which should solve all our issues. Please, change the comment next to a public member to reflect that.